js/menuRenderer.js

```diff
@@ -94,6 +95,58 @@ module.exports = {
       browserUI.addTab(newTab, {
         enterEditMode: !data.url // only enter edit mode if the new tab is empty
       })
+
+      if (data.taskQuery) {
+        // use the first search result
+        // if there is no result, need to create a new task
+        let task
+
+        if (/^\d+$/.test(data.taskQuery)) {
+          task = tasks.get(data.taskQuery)
+        } else {
+          task = searchAndSortTasks(data.taskQuery, excludeSelected=false)[0]?.task
+        }
+
+        if (!task) {
+          task = tasks.get(tasks.add(undefined, tasks.getIndex(tasks.getSelected().id) + 1))
+          task.name = data.taskQuery
+        }
+
+        moveToTaskCommand(task.id)
```

Comment:
IIUC, this is creating a tab in the current task, then using `moveToTaskCommand` to move it to the new task. This should just be a switchToTask call before the addTab call instead.

(However, if we send a switchToTask IPC from the main process, as I suggested in my other comment, this whole block of code should go away anyway).